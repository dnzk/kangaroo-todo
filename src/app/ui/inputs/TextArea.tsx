import style from "./inputs.module.scss"
import font from "@/app/lib/font"

type TextAreaProps = React.HTMLProps<HTMLTextAreaElement> & { invalid?: boolean }

export default function TextArea(props: TextAreaProps) {
  return (
    <textarea className={`${font.className} ${style.base} ${props.invalid ? style.error : ''}`} {...props} />
  )
}
