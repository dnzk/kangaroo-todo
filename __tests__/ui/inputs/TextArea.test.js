import '@testing-library/jest-dom'
import { render } from '@testing-library/react'
import TextArea from '../../../src/app/ui/inputs/TextArea'

describe('TextArea', () => {
  it('renders correctly', () => {
    const { container } = render(<TextArea />)
    expect(container).toMatchSnapshot()
  })
})
