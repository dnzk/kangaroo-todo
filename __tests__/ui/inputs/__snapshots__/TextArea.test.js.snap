// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea renders correctly 1`] = `
<div>
  <textarea
    class="className base "
  />
</div>
`;
